src/__tests__/dialog.spec.tsx

```diff
@@ -0,0 +1,176 @@
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { DialogContent } from "../dialog";
+import React from 'react'
+import '@testing-library/jest-dom'
+import { requestAPI } from '../utils/util';
+import { when } from 'jest-when'
+
+jest.mock("../utils/util", () => ({
+    requestAPI: jest.fn()
+}))
+
+const DEFAULT_VALID_API_KEY = "valid_api_key"
+const DEFAULT_INVALID_API_KEY = "invalid_api_key"
+const DEFAULT_PROJECT_ID = "project_id_abc"
+const DEFAULT_JOB_ID = "job_id_abc"
+
+
+const mockEnvWithAPI = (APIKey: any = null) => {
+    when((requestAPI as jest.Mock)).calledWith("apikey").mockResolvedValue({
+        data: APIKey
+    })
+}
+
+const mockPostAPI = (result: any, api_key: string) => {
+    when((requestAPI as jest.Mock)).calledWith("apikey", { "body": `{\"api_key\":\"${api_key}\"}`, "method": "POST" }).mockResolvedValue({
+        result: result
+    })
+}
+
+// If projectID is provided, the API is used for re-deployment
+const mockJobDeployResult = (projectID = "", deployment_result: object = {
+    project_id: DEFAULT_PROJECT_ID,
+    id: DEFAULT_JOB_ID
+}) => {
+    when((requestAPI as jest.Mock)).calledWith("job", { "body": `{\"api_key\":\"${DEFAULT_VALID_API_KEY}\",\"project_id\":\"${projectID}\"}`, "method": "POST" }).mockResolvedValue({
+        deployment_result: deployment_result
+    })
+}
+
+const renderDialogContent = (project_id: any = null) => {
```

Comment:
what is this function for?

Reply:
The `render(<SomeComponent />)` is to load the test component, this is invoked in every test function so I made a shareable function for this

[Example](https://testing-library.com/docs/react-testing-library/example-intro/)